### DISCUSSION TITLE
How to customize the place of declaration of a class.

### DISCUSSION THREAD

**Discussion by superlopuh** [microsoft/pyright#8716] [Q&A]
This is more of a question about the Python LSP server in VSCode but I'm not sure where the best place to post it is.

I have a directory in a project that I would like to pretend is a file, and to hide the inner folder structure. It has a number of files that are pure implementation detail. How do I export symbols in such a way that VSCode automatically imports them from the public re-export?

``` python
proj/public/_private.py

class _A: pass
```

``` python
proj/public/__init__.py

from ._private import _A as A  # noqa: TID251

# I was hoping this would redirect the LSP but it seems to have 0 effect
__all__ = [
    "A",
]
```

``` python
proj/file.py

# from proj.public import A              <-- this is what I would like the fixit to produce
# from proj.public._private import _A    <-- this is what it currently produces

# Non-imported A
print(A)
```

Is there a way to get this to work as intended, and hide the declaration of things from clients?

**Comment by erictraut** [comment]
The auto-completion logic in the pyright LSP is maintained by the pylance team. I think I remember seeing some issues related to this one in the [pylance-releases issue tracker](https://github.com/microsoft/pylance-release/issues). If not, you could file a bug report or enhancement request there.

**Reply by superlopuh** [reply]
Thank you for the link! Will ask there.

**Reply by superlopuh** [reply]
https://github.com/microsoft/pylance-release/issues/6252
